kb/2023-08-21/compatible_with_ip_pool_new_feature.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Upgrading Guest Clusters with Harvester IP Pool Compatibility 
+description: Explain how to keep load balancer IP during upgrading guest cluster
+slug: upgrading_guest_clusters_with_harvester_ip_pool_compatibility
+authors:
+- name: Canwu Yao
+  title: Software Engineer
+  url: https://github.com/yaocw2020
+  image_url: https://avatars.githubusercontent.com/u/7421463?s=400&v=4
+tags: [harvester, load balancer, cloud provider, ip pool, upgrade] 
+hide_table_of_contents: false
+---
+
+As the **Harvester v1.2.0** releases, a new Harvester cloud provider version **0.2.2** is integrated into the RKE2 **v1.27.3+rke2r1**, **v1.26.6+rke2r1**, **v1.25.11+rke2r1** and **v1.24.15+rke2r1**.
+
+With Harvester v1.2.0, the new Harvester cloud provider provides more powerful load balancing for Kubernetes services. Harvester IP Pool, a built-in IP address management (IPAM) solution for the Harvester load balancer, allows users to provide an IP pool confined to the specific guest cluster by specifying the guest cluster name. For example, the following IP pool is only available for the guest cluster `cluster2`:
+
+![image](ippoolforcluster2.png)
+
+However, the feature is not compatible with the existing guest clusters even after upgrading because they didn't pass the cluster name to the Harvester cloud provider(refer to [issue 4232](https://github.com/harvester/harvester/issues/4232)). As a workaround, users can manually upgrade the Harvester cloud provider using Helm and pass the cluster name after upgrading, but this will change the IPs of the load balancer. This article will provide a workaround to allow users to use the new IP pool feature while keep the load balancer IPs unchanged.
```

Comment:
```suggestion
However, the feature is not automatically compatible with existing guest Kubernetes clusters after upgrading, as they do not pass the correct cluster name to the Harvester cloud provider (refer to [issue 4232](https://github.com/harvester/harvester/issues/4232)). Users can manually upgrade the Harvester cloud provider using Helm as a workaround and provide the correct cluster name after upgrading. However, this would result in a change in the load balancer IPs. This article outlines a workaround that allows users to leverage the new IP pool feature while keeping the load balancer IPs unchanged.
```